app/api/server/v1/voip/extensions.ts

```diff
@@ -48,27 +44,11 @@ API.v1.addRoute('connector.extension.getDetails', { authRequired: true }, {
  */
 API.v1.addRoute('connector.extension.getRegistrationInfo', { authRequired: true }, {
 	get() {
-		const config: IVoipServerConfig = Promise.await(
-			Voip.getServerConfigData(ServerType.CALL_SERVER),
-		) as unknown as IVoipServerConfig;
-		if (!config) {
-			this.logger.warn({ msg: 'API = connector.extension.getRegistrationInfo callserver settings not found' });
-			return API.v1.notFound();
-		}
-		const endpointDetails = Promise.await(commandHandler.executeCommand(
-			Commands.extension_info,
-			this.requestParams(),
-		)) as IVoipConnectorResult;
-		const callServerConfig: ICallServerConfigData = config.configData as unknown as ICallServerConfigData;
-		const endpointInfo: IVoipExtensionConfig = endpointDetails.result as IVoipExtensionConfig;
-		const extensionRegistrationInfo = {
-			sipRegistrar: config.host,
-			websocketUri: callServerConfig.websocketPath,
-			extension: endpointInfo.name,
-			password: endpointInfo.password,
-		};
+		check(this.requestParams(), Match.ObjectIncluding({
+			extension: String,
+		}));
+		const endpointDetails = Promise.await(Voip.getRegistrationInfo(this.requestParams()));
 		this.logger.debug({ msg: 'API = connector.extension.getRegistrationInfo', result: endpointDetails });
-
-		return API.v1.success({ ...extensionRegistrationInfo });
+		return API.v1.success({ ...endpointDetails.result });
```

Comment:
No. In case of correct command execution, the object should be valid (We reject the promise in case of error). Even if there is a glitch (Sudden n/w failure/asterisk crash and so on) we will return an empty object of IVoipConnectorResult having valid (and empty) result parameter. So we are good here.